app/configurator/components/field.tsx

```diff
@@ -252,94 +258,132 @@ export const DataFilterSelect = ({
-          ...dimension.values,
-        ]
-      : dimension.values;
-  }, [isOptional, dimension.values, noneLabel]);
-
-  const allOptionsSet = useMemo(() => {
-    return new Set(
-      allOptions
-        .filter((x) => x.value !== FIELD_VALUE_NONE)
-        .map((x) => {
-          try {
-            return x.value as string;
-          } catch (e) {
-            console.warn(`Bad value ${x.value}`);
-            return;
-          }
-        })
-        .filter(truthy)
-    );
-  }, [allOptions]);
-
-  const isDisabled = useCallback(
+  const usesMostRecentDate = isDynamicMaxValue(fieldProps.value);
+  const label = isOptional ? (
+    <>
+      {_label}{" "}
+      <span style={{ marginLeft: "0.25rem" }}>
+        (<Trans id="controls.select.optional">optional</Trans>)
+      </span>
+    </>
+  ) : (
+    _label
+  );
+  const { minDate, maxDate, optionValues } = React.useMemo(() => {
+    if (values.length) {
+      const options = values.map((d) => {
+        return {
+          label: `${d.value}`,
+          value: `${d.value}`,
+        };
+      });
+
+      return {
+        minDate: parseDate(`${values[0].value}`) as Date,
+        maxDate: parseDate(`${values[values.length - 1].value}`) as Date,
+        optionValues: options.map((d) => d.value),
+      };
+    } else {
+      const date = new Date();
+      return {
+        minDate: date,
+        maxDate: date,
+        optionValues: [],
+      };
+    }
+  }, [values, parseDate]);
+  const isDateDisabled = React.useCallback(
     (date: Date) => {
-      return !allOptionsSet.has(formatDate(date));
+      return !optionValues.includes(formatDate(date));
     },
-    [allOptionsSet]
+    [optionValues, formatDate]
   );
 
-  const dateValue = useMemo(() => {
-    const parsed = fieldProps.value ? parseDate(fieldProps.value) : undefined;
-    return parsed || new Date();
-  }, [fieldProps.value]);
-
-  const [minDate, maxDate] = useMemo(() => {
-    const [min, max] = extent(Array.from(allOptionsSet));
-    if (!min || !max) {
-      return [];
-    }
-    return [new Date(min), new Date(max)] as const;
-  }, [allOptionsSet]);
-
   return (
     <DatePickerField
-      label={isOptional ? `${label} (${optionalLabel})` : label}
-      disabled={disabled}
-      controls={controls}
+      name={`date-picker-${dimension.iri}`}
+      label={
+        <Flex
+          sx={{
+            width: "100%",
+            justifyContent: "space-between",
+            alignItems: "center",
+          }}
+        >
+          <FieldLabel
+            label={
+              <OpenMetadataPanelWrapper dim={dimension}>
+                {label}
+              </OpenMetadataPanelWrapper>
+            }
+          />
+          {/* FIXME: adapt to design */}
+          <FormGroup>
+            <FormControlLabel
+              control={
+                <MUISwitch
+                  checked={usesMostRecentDate}
+                  onChange={() =>
+                    fieldProps.onChange({
```

Comment:
Nit: Instead of relying on the state usesMOstRecentDate, I think it is preferrable to rely on the `checked` value coming in the 2nd argument of the callback. This way the function could be stable, and I think it'd be clearer since here you have to think "ok usesMostRecentDate is true, so when it is clicked it shuold be false, and the max date should be used". IN the callback, the checked parameter is already right, so I think it's clearer.